### DISCUSSION TITLE
Is there a function that returns rows as a List<Map<String, dynamic>> ?

### DISCUSSION THREAD

**Discussion by allaboutmikey** [isoos/postgresql-dart#127] [Q&A] [answered]
The function mappedResultsQuery returns a `List<Map<String, Map<String, dynamic>>>` which I find really awkward to access:
```
var tID = row["tablename"]["id"];
```
In other languages I usually work with database rows as a dictionary. This translates to a `List<Map<String, dynamic>>` IIUC.
Then the access is:
```
var tID = row["id"]
```
I find this makes it easier when putting/getting values into/from UI elements. 
Am I missing a function or can this not be done?

**Comment by isoos** [comment, marked as answer]
https://pub.dev/documentation/postgres/latest/postgres/PostgreSQLExecutionContext/query.html
the result is a list of rows, which can be accessed by index, and also via a few methods:
https://pub.dev/documentation/postgres/latest/postgres/PostgreSQLResultRow-class.html
https://pub.dev/documentation/postgres/latest/postgres/PostgreSQLResultRow/toColumnMap.html

**Reply by allaboutmikey** [reply]
Thanks @isoos 
The toColumnMap method does what I was looking for, for the individual result row. I can get by with that.
It seems there is no function that returns a list of rows already in this format.

**Reply by isoos** [reply]
`row.map((e)=>e.toColumnMap()).toList()`

**Reply by allaboutmikey** [reply]
@isoos I must have applied what you've written incorrectly. It doesn't seem to change anything.
To clarify, I was able to use **toColumnMap** in order to access fields from each row by name:
```
import 'package:postgres/postgres.dart';

void main() async {
  var connection = PostgreSQLConnection("192.168.x.x", 5432, "dbname",
      username: "uname", password: "pword");
  await connection.open();

  PostgreSQLResult results = await connection.query(
      "SELECT bkid, bookeddate, paxname, company FROM bookings b join agents a on b.agentcode=a.code WHERE canx = @aValue",
      substitutionValues: {"aValue": 1});

  for (final row in results) {
    var newrow = row.toColumnMap();
    print('ID:${newrow['bkid']}: ${newrow['bkid']} : ${newrow['paxname']}');
  }
}
```
I tried using your suggested `row.map((e)=>e.toColumnMap()).toList()` on the PostgresqlSQLResult, but it didn't seem to change anything when I looked at the resulting rows in the debugger. I had a list of lists, not a list of maps. I thought removing the final **toList** might do the trick, but obviously I'm still too new to rust to know what I'm talking about, because that didn't do what I thought it would either.

What I usually do in Python, for example, would translate (I believe) into something like this:
```
import 'package:postgres/postgres.dart';

void main() async {
  var connection = PostgreSQLConnection("192.168.x.x", 5432, "dbname",
      username: "uname", password: "pword");
  await connection.open();

  List<Map<String,dynamic>> results = await connection.columnMappedResultsQuery(
      "SELECT bkid, bookeddate, paxname, company FROM bookings b join agents a on b.agentcode=a.code WHERE canx = @aValue",
      substitutionValues: {"aValue": 1});

  for (final row in results) {
    print('ID:${row['bkid']}: ${row['bkid']} : ${row['paxname']}');
  }
}
```
It seems the **mappedResultsQuery** function applies **toTableColumnMap** to each row in the list. Perhaps it should have been called **tableMappedResultsQuery**.
I'd like the option to have **toColumnMap** applied to each row instead, which is what my imagined function **columnMappedResultsQuery** would do.
I hope that's clearer.
